contracts/PLCR.sol

```diff
@@ -0,0 +1,398 @@
+        uint64 _revealDuration
+    )
+        onlyInit
+        external
+    {
+        initialized();
+
+        require(isContract(_staking));
+        require(_voteQuorum <= PCT_BASE);
+        require(_minorityBlocSlash <= PCT_BASE);
+
+        staking = _staking;
+        voteQuorum = _voteQuorum;
+        minorityBlocSlash = _minorityBlocSlash;
+        commitDuration = _commitDuration;
+        revealDuration = _revealDuration;
+    }
+
+    /**
+     * @notice Create a new vote about "`_metadata`"
+     * @param _script EVM script to be executed on approval
+     * @param _metadata Vote metadata
+     * @return voteId Id for newly created vote
+     */
+    function newVote(bytes _script, string _metadata) isInitialized external auth(CREATE_VOTE_ROLE) returns (uint256 voteId) {
+        voteId = votes.length++;
+
+        Vote storage vote = votes[voteId];
+        vote.commitEndDate = getTimestamp().add(commitDuration);
+        vote.revealEndDate = vote.commitEndDate.add(revealDuration);
+        vote.metadata = _metadata;
+
+        NewVote(voteId);
+        return voteId;
+    }
+
+    /**
+     * @notice Commit to vote `_voteId` using lock with id `_lockId` for option contained in hash `_secretHash`
+     * @param _voteId Id of the Vote
+     * @param _secretHash Hash obtained from own salt and voting option
+     * @param _lockId Id of the lock from Staking app used for this vote
+     */
+    function commitVote(uint256 _voteId, bytes32 _secretHash, uint256 _lockId) isInitialized public {
+        Vote storage vote = votes[_voteId];
+
+        // check commit period for Vote
+        require(getTimestamp() <= vote.commitEndDate);
+
+        // check lock and get amount
+        uint256 stake = checkLock(msg.sender, _lockId, vote.commitEndDate, vote.revealEndDate);
+
+        // move slash proportion to here
+        uint256 slashStake = stake.mul(minorityBlocSlash) / PCT_BASE;
+        staking.unlockPartialAndMoveTokens(msg.sender, _lockId, address(this), slashStake);
+        vote.slashPool = vote.slashPool.add(slashStake);
+
+        vote.userVotes[msg.sender] = UserVote({
+            lockId: _lockId,
+            stake: stake,
+            secretHash: _secretHash,
+            revealed: false,
+            voteOption: false,
+            claimed: false
+        });
+
+        CommitedVote(_voteId, msg.sender, stake, _secretHash);
+    }
+
+    /**
+     * @notice Reveal option `_voteOption` for Vote `_voteId` with salt `_salt`
+     * @param _voteId Id of the Vote
+     * @param _voteOption Option that was preciously commited for
+     * @param _salt Salt used in previous vote commit
+     */
+    function revealVote(uint256 _voteId, bool _voteOption, bytes32 _salt) isInitialized public {
+        Vote storage vote = votes[_voteId];
+        UserVote storage userVote = votes[_voteId].userVotes[msg.sender];
+
+        // check reveal period
+        require(vote.commitEndDate < getTimestamp());
```

Comment:
Seems like we could aggregate these into a `commitPeriodActive` bool on `UserVote`